File: leet_SLWD_2024_max_confuse.go
func maxConsecutiveAnswers(s string, k int) int {
    len := len(s)
    res := 0
    t := 0
    f := 0
    l := 0
    r := -1
    for r < len - 1 {
        r++
        if s[r] == 'T' {
            t++
        } else {
            f++
        }
        for t > k && f > k {
            if s[l] == 'T' {
                t--
            } else {
                f--
            }
            l++
        }
        if r - l + 1 > res {
            res = r - l + 1
        }
    }
    return res
}

File: leet_SLWD_2024_max_confuse.cc
class Solution {
public:
    int maxConsecutiveAnswers(string s /* answerKey */, int k)
    {
        int res = 0;
        int len = s.length();
        int T = 0, F = 0;
        int L = 0;
        int R = -1;

        while (++R < len)
        {
            if (s[R] == 'T')
                ++T;
            else
                ++F;
            while (T > k && F > k)
            {
                if (s[L] == 'T')
                    --T;
                else
                    --F;
                ++L;
            }
            res = max(res, R - L + 1);
        }
        return res;
    }
};
